src/rez/pip.py

```diff
@@ -315,11 +318,10 @@ def make_root(variant, path):
 
         variant_reqs.append("python-%s" % py_ver)
 
-        name, _ = parse_name_and_version(distribution.name_and_version)
-        name = distribution.name[0:len(name)].replace("-", "_")
+        name = distribution.metadata.name
 
         with make_package(name, packages_path, make_root=make_root) as pkg:
-            pkg.version = distribution.version
+            pkg.version = distribution.metadata.version
```

Comment:
Note that we need to be careful with this. The python version schema specification isn't 100% compatible with rez.

Reply:
I have been using metadata in a couple of places for summary and name, those shouldn't be an issue I guess. I never had an issue with version but I can revert back to the original if necessary.

This is an issue I have been getting but I fixed it on the other PR

https://github.com/nerdvegas/rez/pull/628/commits/98be1156cc61d54d846304f13028d6bcfeb0e369

Reply:
I'd second JC here, even if initially there's no remapping at all, there should be a separate `pip_to_rez_version` function.

Taking a look at https://www.python.org/dev/peps/pep-0440/#version-scheme though, there are some things to take into account. I can see two potential issues
1: version epochs (they make no sense to rez, so they'd just get stripped of the leading `N!`;
2: python versions are case insensitive, so they should probably be lowercased when converted to a rez version.

Reply:
Local versions are also not compatible with rez (https://www.python.org/dev/peps/pep-0440/#local-version-segments) BTW, local is something we use at RodeoFX for our pip packages when we fork a repo for example. Though, we don't use `rez-pip` since we have our own implementation. But some other studios could use the local versions. At Rodeo, we basically convert `package-1.0.0+<something>` to `package-1.0.0-<something>`.

Most possibilities are described in https://www.python.org/dev/peps/pep-0440/#examples-of-compliant-version-schemes.